File: c++/water_trap.cpp
#include <iostream>

using namespace std;

int main(){
  
  int n, i, sum = 0, x = 0, y = 0;
  cin >> n;
  
  int blocks[n];
  for (i=0; i<n; i++) cin >> blocks[i];
  
  for (i=0; i<n; i++) {
    x = max(x, blocks[i]);
    sum += x - blocks[i];
  }
  
  for (i=n-1; blocks[i] != x; i--) {
    y = max(y, blocks[i]);
    sum -= x - y;
  }
  
  cout << sum;
  
  return 0;
}

File: Python/Water_Trap.py
a=[int(i) for i in input().split()]
b=a.copy()
l=len(a)
print(b[0],end=' ')
for i in range(1,l-1):
    be=max(b[0:i])
    af=max(b[i+1:])
    if (be<af):
        t=be
    else:
        t=af
    if (b[i]<t):
        b[i]=t
    print(b[i],end=' ')
print(b[-1])
print(abs(sum(a)-sum(b)))